src/panels/lovelace/cards/hui-button-card.ts

```diff
@@ -71,10 +77,39 @@ export class HuiButtonCard extends LitElement implements LovelaceCard {
     };
   }
 
-  @property({ attribute: false }) public hass?: HomeAssistant;
+  public hass!: HomeAssistant;
 
   @state() private _config?: ButtonCardConfig;
 
+  @consume<any>({ context: statesContext, subscribe: true })
+  @transform({
+    transformer: function (this: HuiButtonCard, value: HassEntities) {
+      return this._config?.entity ? value[this._config?.entity] : undefined;
+    },
+    watch: ["_config"],
+  })
+  _state?: HassEntity;
```

Comment:
```suggestion
  _stateObj?: HassEntity;
```
To be consistent with naming in other files.